src/cluster_legacy.c

```diff
@@ -2810,6 +2892,18 @@ int clusterIsValidPacket(clusterLink *link) {
         explen = sizeof(clusterMsg) - sizeof(union clusterMsgData);
         explen += sizeof(clusterMsgDataPublish) - 8 + ntohl(hdr->data.publish.msg.channel_len) +
                   ntohl(hdr->data.publish.msg.message_len);
+    } else if (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT) {
+        clusterMsgLight *hdr_pubsub = (clusterMsgLight *)link->rcvbuf;
+        explen = sizeof(clusterMsgLight) - sizeof(union clusterMsgDataLight);
+        explen += sizeof(clusterMsgDataPublishMulti);
+        uint64_t data_count = ntohu64(hdr_pubsub->data.publish.msg.data_count);
+        explen += ((data_count) * (sizeof(clusterMsgDataPublishMessage) - 8));
```

Comment:
Let's not have magic number. Can we determine the value `8` programmatically?

Reply:
This seems also not addressed and resolved. I think it helps understand easier when computed programmatically rather than hardcoded. 

If it's not clean, I would suggest adding code comment referring to the placeholder size.

Reply:
I remember adding it in some place, might have reverted it. I will re-add

Reply:
I added the change in the create function and I resolved this 
`(sizeof(((clusterMsgDataPublishMessage *)0)->message_data))))`